### ISSUE TITLE
reposync symbolic links awareness 

### ISSUE THREAD

**Issue by H0neyBadger** [rpm-software-management/yum-utils#46]
Hello, 
I regularly use symlinks to save some space on sub-repositories.
Is it possible to add the following changes to avoid symlinks deletion during the reposync ?

```diff
diff --git a/reposync b/reposync
index eb03c7d..562c3c9 100755
--- a/reposync
+++ b/reposync
@@ -80,13 +80,13 @@ def localpkgs(directory):
             subcache = localpkgs(fn)
             for pkg in subcache.keys():
                 cache[pkg] = subcache[pkg]
-        elif stat.S_ISREG(st.st_mode) and name.endswith(".rpm"):
+        elif (stat.S_ISREG(st.st_mode) or stat.S_ISLNK(st.st_mode)) and name.endswith(".rpm"):
             cache[name] = { 'path': fn, 'size': st.st_size, 'device': st.st_dev }
     return cache
 
 def is_subpath(path, root):
     root = os.path.realpath(root)
-    path = os.path.realpath(os.path.join(root, path))
+    path = os.path.normpath(os.path.join(root, path))
     # join() is used below to ensure root ends with a slash
     return path.startswith(os.path.join(root, ''))
```

**Comment by james-antill** [comment]
The usual way to do this is to use hardlinks, often by running the hardlink command. Is there a reason you prefer symlinks?

Also the is_subpath change seems wrong, given how each of those functions work (with symlinks to directories).

**Comment by H0neyBadger** [comment]
Thank you a lot for your analysis.
You are right about the hardlink I will give it a try. 

Regarding the symlink, I used *normpath* to resolve the path without translating the symlink
 But it looks inconsistent:
> This string manipulation may change the meaning of a path that contains symbolic links.

I tried on two different servers and I got different results.
I am closing this issue since I will use the hardlink instead.
Thank you very much for your help !
